### ISSUE TITLE
Dialog dismiss: View not attached to window manager (Android)

### ISSUE THREAD

**Issue by PaulAchess** [aritchie/userdialogs#13]
Hi, I hope you're not starting to hate me.

I had a strange error in production that was reproduced a few times, by only one user so far.
The device used was a GT-N7100 (Galaxy Note 2). My report manager tells me the application is backgrounded.

During the dismiss of a dialog (a loading to be precise), I have on a few occasions a « RuntimeException, View=… not attached to window manager ».

From what I see here:
http://stackoverflow.com/questions/13103237/android-rare-runtime-exception-that-does-not-point-to-any-class-where-it-start
http://stackoverflow.com/questions/2224676/android-view-not-attached-to-window-manager

This seems to be fixable by checking if the dialog can actually be dismissed. I don't know if this should be reported to AndHUD or to you though, as it seems to be a lower level exception.
It's probably coming from the samsung ROM, as this touches the activity lifecycle.

Full stacktrace : 
![stacktrace_viewnotattached](https://cloud.githubusercontent.com/assets/6310023/6706590/43142156-cd5f-11e4-8d95-36a837fc7a0a.png)

**Comment by aritchie** [comment]
Paul - I am beginning to feel like a bit of a front line support guy for you.  You need to supply actual code and/or full stack traces or better yet, actual fixes for the problems you're experiencing.  I use these libraries in many of my projects and I don't get any of the issues you seem to be experiencing.

**Comment by PaulAchess** [comment]
Sorry – this was not my intention at all, believe me. I'm grateful for your work and my goal was only to provide informations on the crashes I find. I found nothing really usefull on this error on the web, I felt that posting this could help gather people who might encounter the same issue.

I'll try my best to get a reproduction scenario / test fixes on my side. Only one user is experiencing this, so I'll be able to test fixes. In my opinion there might be something odd on the application lifecycle of this ROM and I believe the use of async on my side is messing with that.

**Comment by PaulAchess** [comment]
Hi, I think I found something odd that might explain this crash (by the way, several users are experiencing this now).

When looking at the code of `ProgressDialog`, I found that the `Refresh` uses the following code :

    Utils.RequestMainThread(() => AndHUD.Shared.Show(
                ActivityMonitor.CurrentTopActivity, 
                txt,
                p, 
                MaskType.Black,
                null,
                this.OnCancelClick
            ));

and the `Hide` uses this one:

    public virtual void Hide() {
            this.IsShowing = false;
            Utils.RequestMainThread(() => AndHUD.Shared.Dismiss(ActivityMonitor.CurrentTopActivity));
        }

But can we be sure that the « ActivityMonitor.CurrentTopActivity » is 
1- the same between Hide & Show ?
2- the correct one to use on Hide ?

I'm not an expert at all, so I'm just raising the question here. But considering all other dialogs use `this.getTopActivity`, I think this could explain the crash.


I also looked at the old package (on which I never had the error), the context used on ProgressDialog was `Utils.GetActivityContext()` for everyone. I see you refactored all this but I don't get why ProgressDialog uses a different method to get the context.

Hoping this helps, and again sorry if you feel like a front line support, this is absolutely not the case nor my intention (currently, I have a try-catch to prevent this crash, so this is not blocking nor urgent).
